File: src/cymbalism-accelerando.hs
-- cymbalism accellerando (jmcc)

import Sound.SC3.ID
import Sound.SC3.Lang.Events.OverlapTexture

enumFromN :: Enum a => a -> Int -> [Int]
enumFromN e i = let j = fromEnum e in [j .. j + i]

cymbalism_accellerando :: UGen
cymbalism_accellerando =
  let y n = let f1 = rand n 500 2500
                f2 = rand n 0 8000
                f = map (\e -> rand e f1 (f1 + f2)) (enumFromN n 15)
                rt = map (\e -> rand e 1 5) (enumFromN n 15)
          in klankSpec f (replicate 15 1) rt
      z = mce2 (y 'a') (y 'A')
      w = whiteNoise 'α' AR * 0.02
      tf = xLine KR (linRand 'β' 0.5 4.5 0) (rand 'γ' 0.5 35.5) 12 DoNothing
      t = impulse ar tf 0
      s = decay t 0.004 * w
  in klank s 1 0 1 (mceTranspose z)

main :: IO ()
main = overlapTextureU (4,4,4,maxBound) cymbalism_accellerando

File: gr/jmcc/cymbalism-accelerando.fs
\ cymbalism accelerando (jmcc) #2

: n 15 ;
: f1 500 2500 Rand.ir ;
: f2 0 8000 Rand.ir ;
: param n 0 do f1 f1 f2 + Rand.ir 1 1 5 Rand.ir loop n 3 * mce ;
: tf 0.5 4.5 0 LinRand.ir 0.5 35.5 Rand.ir 12 0 XLine.kr ;
tf 0 Impulse.ar 0.004 Decay WhiteNoise.ar 0.02 * * 1 0 1 param Klank.ar 2 clone
4 4 inf xfade-texture
